src/base/bittorrent/session.cpp

```diff
@@ -1405,9 +1405,14 @@ bool Session::deleteTorrent(const QString &hash, bool deleteLocalFiles)
 
     // Remove it from session
     if (deleteLocalFiles) {
-        QString rootPath = torrent->rootPath(true);
-        if (!rootPath.isEmpty())
-            m_savePathsToRemove[torrent->hash()] = rootPath;
+        if (torrent->savePath(true) == torrentTempPath(torrent->hash())) {
+            m_savePathsToRemove[torrent->hash()] = torrent->savePath(true);
+        }
+        else {
+            QString rootPath = torrent->rootPath(true);
+            if (!rootPath.isEmpty())
+                m_savePathsToRemove[torrent->hash()] = rootPath;
+        }
         m_nativeSession->remove_torrent(torrent->nativeHandle(), libt::session::delete_files);
     }
     else {
```

Comment:
>I thought about it... but I decided to wait for your feedback first. 

Lets keep the current way for now, so it won't break the following usage:
A user is downloading a mutil-files torrent, the download is incomplete/partial complete, the users previewed the contents and think "OK I got what I need, I'm going to remove the download job and keep the files".